pkg/framework/plugins/removepodsviolatinginterpodantiaffinity/pod_antiaffinity.go

```diff
@@ -78,24 +78,36 @@ func (d *RemovePodsViolatingInterPodAntiAffinity) Name() string {
 }
 
 func (d *RemovePodsViolatingInterPodAntiAffinity) Deschedule(ctx context.Context, nodes []*v1.Node) *framework.Status {
+	podsList, err := d.handle.ClientSet().CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return &framework.Status{
+			Err: fmt.Errorf("error listing all pods: %v", err),
+		}
+	}
+
+	podsInANamespace := groupByNamespace(podsList)
+
+	runningPodFilter := func(pod *v1.Pod) bool {
+		return pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed
+	}
```

Comment:
Also, I'm afraid that this code duplicating with following:

https://github.com/kubernetes-sigs/descheduler/blob/93a014ef5ebe66d4945f5ad181deb5af9c61ccd8/pkg/descheduler/pod/pods.go#L170-L172